### PR TITLE
Adds StaffAndCourseCreatorOnly mixin and relevant helper methods/tests

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
Adds the following:
- Adds `StaffAndCourseCreatorOnly` mixin to restrict access based on course creator role of user
- Adss helper methods and tests

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-1412

### CODE REVIEW

File: ecommerce/core/views.py [hunk 1]
```diff
@@ -104,6 +105,16 @@ def dispatch(self, request, *args, **kwargs):
 
         return super(StaffOnlyMixin, self).dispatch(request, *args, **kwargs)
 
+class StaffOrCourseCreatorOnlyMixin(object):
```
Comment (zubair-arbi) [inline ecommerce/core/views.py:108]: Move this mixin to ` ecommerce/extensions/edly_ecommerce_app/views.py`

File: ecommerce/core/views.py [hunk 2]
```diff
@@ -104,6 +105,16 @@ def dispatch(self, request, *args, **kwargs):
...
+    """ Makes sure only staff users and course creators can access the view. """
```
Comment (zubair-arbi) [inline ecommerce/core/views.py:109]: update docstring https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

File: ecommerce/extensions/edly_ecommerce_app/helpers.py [hunk 3]
```diff
@@ -48,6 +48,23 @@ def get_edx_org_from_edly_cookie(encoded_cookie_data):
...
 
+def get_is_course_creator_from_edly_cookie(encoded_cookie_data):
+    """
+    Returns "is_course_creator" value from the "edly-user-info" cookie.
+
+    Arguments:
+        encoded_cookie_data (dict): Edly user info cookie JWT encoded string.
+
+    Returns:
+        string
```
Comment (zubair-arbi) [inline ecommerce/extensions/edly_ecommerce_app/helpers.py:59]: boolean?

File: ecommerce/core/views.py (hunk 1 above)
Comment (rehan99000) [reply ecommerce/core/views.py:108]: previous views are in `api/v1/views` so I have moved it there

File: ecommerce/core/views.py (hunk 2 above)
Comment (rehan99000) [reply ecommerce/core/views.py:109]: `StaffOnlyMixin` had the docstring in the same style so I did it the same way.

File: ecommerce/extensions/edly_ecommerce_app/helpers.py (hunk 3 above)
Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py:59]: changed

File: ecommerce/extensions/edly_ecommerce_app/helpers.py [hunk 4]
```diff
@@ -48,6 +48,23 @@ def get_edx_org_from_edly_cookie(encoded_cookie_data):
...
 
+def get_is_course_creator_from_edly_cookie(encoded_cookie_data):
+    """
+    Returns "is_course_creator" value from the "edly-user-info" cookie.
+
+    Arguments:
+        encoded_cookie_data (dict): Edly user info cookie JWT encoded string.
+
+    Returns:
+        boolean
```
Comment (zubair-arbi) [inline ecommerce/extensions/edly_ecommerce_app/helpers.py:59]: nit: be consistent with `bool` https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

File: ecommerce/extensions/edly_ecommerce_app/helpers.py [hunk 5]
```diff
@@ -92,3 +109,16 @@ def user_has_edly_organization_access(request):
     edx_org_short_name = get_edx_org_from_edly_cookie(edly_user_info_cookie)
 
     return partner_short_code == edx_org_short_name
+
+def user_is_course_creator(request):
```
Comment (ziafazal) [inline ecommerce/extensions/edly_ecommerce_app/helpers.py:113]: nit: I think we should merge this and `get_is_course_creator_from_edly_cookie` to make is more simple single method and name it `user_is_course_creator`.

Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py:96]: Done

File: ecommerce/extensions/edly_ecommerce_app/helpers.py (hunk 4 above)
Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py]: removed this method on @ziafazal recommendation

**Pull request merged**